pkg/host/machine_info_linux.go

```diff
@@ -125,26 +125,55 @@ func readKVMInfo(buffer *bytes.Buffer) error {
 	return nil
 }
 
+func getModuleTextAddr(moduleName string) (uint64, error) {
+	addrPath := filepath.Join("/sys", "module", moduleName, "sections", ".text")
+	addrContent, err := os.ReadFile(addrPath)
+	if err != nil {
+		return 0, fmt.Errorf("could not read module .text address file: %w", err)
+	}
+	addrString := strings.TrimSpace(string(addrContent))
+	addr, err := strconv.ParseUint(addrString, 0, 64)
+	if err != nil {
+		return 0, fmt.Errorf("address parsing error in %v: %w", moduleName, err)
+	}
+	return addr, nil
+}
+
 func getModulesInfo() ([]KernelModule, error) {
 	modulesText, _ := os.ReadFile("/proc/modules")
-	return parseModules(modulesText)
+	modules, err := parseModules(modulesText)
+	if err != nil {
+		return modules, err
+	}
+	// Fix up module addresses to use .text addresses where available.
+	for i, module := range modules {
+		addr, err := getModuleTextAddr(module.Name)
+		if err == nil {
+			modules[i].Addr = addr
+		}
+	}
+	return modules, nil
 }
 
 func parseModules(modulesText []byte) ([]KernelModule, error) {
 	var modules []KernelModule
 	re := regexp.MustCompile(`(\w+) ([0-9]+) .*(0[x|X][a-fA-F0-9]+)[^\n]*`)
 	for _, m := range re.FindAllSubmatch(modulesText, -1) {
-		addr, err := strconv.ParseUint(string(m[3]), 0, 64)
+		name := string(m[1])
+		addr, err := getModuleTextAddr(name)
 		if err != nil {
-			return nil, fmt.Errorf("address parsing error in /proc/modules: %w", err)
+			// Module address unavailable, .text is probably 0. Skip this module.
+			continue
 		}
 		size, err := strconv.ParseUint(string(m[2]), 0, 64)
 		if err != nil {
+			// /proc/modules is broken, bail out.
 			return nil, fmt.Errorf("module size parsing error in /proc/modules: %w", err)
 		}
 		modules = append(modules, KernelModule{
 			Name: string(m[1]),
 			Addr: addr,
+			// Actually the whole size allocated for the module. Will not be used.
```

Comment:
Size should still be used when canonicalizing addresses (https://github.com/google/syzkaller/blob/master/pkg/cover/canonicalizer.go#L99)

Reply:
Hmm, you are right. We can adjust the size by the difference between the module load address and the .text address then.

Reply:
Done.